src/systems/physics/Physics.cc

```diff
@@ -664,36 +710,97 @@ void PhysicsPrivate::CreatePhysicsEntities(const EntityComponentManager &_ecm)
-        {
-          ignwarn << "Model's parent entity [" << _parent->Data()
-                  << "] not found on world map." << std::endl;
-          return true;
-        }
-        auto worldPtrPhys = this->entityWorldMap.at(_parent->Data());
-
+        // Check if parent world / model exists
         sdf::Model model;
         model.SetName(_name->Data());
         model.SetRawPose(_pose->Data());
-
         auto staticComp = _ecm.Component<components::Static>(_entity);
         if (staticComp && staticComp->Data())
         {
           model.SetStatic(staticComp->Data());
         }
-
         auto selfCollideComp = _ecm.Component<components::SelfCollide>(_entity);
         if (selfCollideComp && selfCollideComp ->Data())
         {
           model.SetSelfCollide(selfCollideComp->Data());
         }
 
-        auto modelPtrPhys = worldPtrPhys->ConstructModel(model);
-        this->entityModelMap.insert(std::make_pair(_entity, modelPtrPhys));
+        // check if parent is a world
+        auto worldIt = this->entityWorldMap.find(_parent->Data());
+        if (worldIt != this->entityWorldMap.end())
+        {
+          auto worldPtrPhys = worldIt->second;
+
+          // Use the ConstructNestedModel feature for nested models
+          if (model.ModelCount() > 0)
+          {
+            auto nestedModelFeature = entityCast(_parent->Data(), worldPtrPhys,
+                this->entityWorldNestedModelMap);
+            if (!nestedModelFeature)
+            {
+              igndbg << "Attempting to construct nested models, but the physics"
+                     << " engine doesn't support feature "
+                     << "[ConstructSdfNestedModelFeature]. "
+                     << "Nested model will be ignored."
+                     << std::endl;
+              return true;
+            }
+            auto modelPtrPhys = nestedModelFeature->ConstructNestedModel(model);
+            this->entityModelMap.insert(std::make_pair(_entity, modelPtrPhys));
+          }
+          else
+          {
+            auto modelPtrPhys = worldPtrPhys->ConstructModel(model);
+            this->entityModelMap.insert(std::make_pair(_entity, modelPtrPhys));
+          }
+        }
+        // check if parent is a model (nested model)
+        else
+        {
+          auto parentIt = this->entityModelMap.find(_parent->Data());
+          if (parentIt != this->entityModelMap.end())
+          {
+            auto parentPtrPhys = parentIt->second;
+
+            auto nestedModelFeature = entityCast(_parent->Data(), parentPtrPhys,
+                this->entityModelNestedModelMap);
+            if (!nestedModelFeature)
+            {
+              igndbg << "Attempting to construct nested models, but the physics"
+                     << " engine doesn't support feature "
+                     << "[ConstructSdfNestedModelFeature]. "
+                     << "Nested model will be ignored."
+                     << std::endl;
```

Comment:
If there's a chance this will spam the console, consider using the `static bool informed` pattern that's used in other parts of this file.

Reply:
done. a71a3d2e2e65bab10243987f91ed238a1329f7ab